src/nethsec/objects/__init__.py

```diff
@@ -241,19 +266,25 @@ def add_domain_set(uci, name: str, family: str, domains: list[str], timeout: int
         raise utils.ValidationError('family', 'invalid_family', family)
     if timeout < 0:
         raise utils.ValidationError('timeout', 'invalid_timeout', timeout)
+    if get_domain_set_total_chars(name, domains) > 1024 - 10:  # -10 chars for safety
+        raise utils.ValidationError('domain', 'total_chars_exceeded')
+
+    # remove duplicated domains
+    unique_domains = list(dict.fromkeys(domains))
+
```

Comment:
Just execute unique once and pass them to the function.

Do the same inside the edit function below.
```suggestion
    # remove duplicated domains
    unique_domains = list(dict.fromkeys(domains))
    
    if get_domain_set_total_chars(name, unique_domains) > 1024 - 10:  # -10 chars for safety
        raise utils.ValidationError('domains', 'total_chars_exceeded')

```